src/ParametersHandler/YarpImplementation/include/BipedalLocomotion/ParametersHandler/YarpImplementation.h

```diff
@@ -223,14 +223,22 @@ class YarpImplementation : public IParametersHandler
      * @return A pointer to IParametersHandler, if the group is not found the weak pointer cannot
      * be locked
      */
-    weak_ptr getGroup(const std::string& name) const final;
+    const_weak_ptr getGroup(const std::string& name) const final;
+
+    /**
+     * Get a Group from the handler.
+     * @param name name of the group
+     * @return A pointer to IParametersHandler, if the group is not found the weak pointer cannot
+     * be locked
+     */
+    weak_ptr getGroup(const std::string& name) final;
 
     /**
      * Set a new group on the handler.
      * @param name name of the group
      * @param newGroup shared pointer to the new group
      */
-    bool setGroup(const std::string& name, shared_ptr newGroup) final;
+    bool setGroup(const std::string& name, const_shared_ptr newGroup) final;
```

Comment:
This modification is missing in the TOML implementation.

Also, I would document that the input group is copied, hence the modifications to the original pointer will not have any effect.